python/lsst/daf/butler/core/datastore.py

```diff
@@ -27,13 +27,14 @@
 from lsst.daf.butler.core.utils import doImport
 
 from abc import ABCMeta, abstractmethod
-from .config import Config
+from .config import ConfigSubset
 
 __all__ = ("DatastoreConfig", "Datastore")
 
 
-class DatastoreConfig(Config):
-    pass
+class DatastoreConfig(ConfigSubset):
+    component = "datastore"
+    requiredKeys = ("cls",)
```

Comment:
Isn't this check now duplicated in `Butler(Config)`?

Reply:
ButlerConfig doesn't inherit from ConfigSubset so no. I could move the validation code in `ConfigSubset.__init__()` to `Config.__init__()` and always have it as an option if you set `requiredKeys`.

Reply:
That sounds good.